### DISCUSSION TITLE
Automatically Tag Emails

### DISCUSSION THREAD

**Discussion by austin3410** [the-djmaze/snappymail#1628] [Q&A]
I'm wondering if it's possible to create rules to automatically apply certain tags based on a set of criteria.

In my specific use case, I have multiple addresses going to one mailbox (user@example.com, and user@example2.com) emails from both of those addresses go into user@example.com mailbox. Can I automatically tag any emails that come from user@example2.com? That way I can very quickly distinguish which address that particular email came from?

I'm trying to recreate Gmails tagging features within Snappymail.

I saw that SIEVE scripts are a thing, but I've never used them and have no idea how to enable them.

If this isn't supported, is there an extension or workaround that would work?

**Comment by the-djmaze** [comment]
When your IMAP server supports Sieve and custom tags, you can do it that way.
There is no other way to automatically apply tags.

Here's an example script
```
require ["imap4flags"];

if header :contains ["To", "CC"] "user@example.com"
{
    addflag "Example";
}

if header :contains ["To", "CC"] "user@example2.com"
{
    addflag "Example2";
}
```
